test/e2e/providers/vsphere/vsphere.go

```diff
@@ -88,7 +127,7 @@ func (p *Provider) GenerateMachineSet(withWindowsLabel bool, replicas int32) (*m
 	}
 
 	// create new machine provider spec for deploying Windows node
-	providerSpec, err := newVSphereMachineProviderSpec(clusterID)
+	providerSpec, err := p.newVSphereMachineProviderSpec()
```

Comment:
I'm not sure about making this a method and no longer passing in `clusterID`, seems like this is going to result in an extra API call.

Reply:
My rationale for that was that the cluster ID is no longer required in that method.  Since we are importing the workspace from an existing machineset, we no longer need the cluster ID to produce a valid workspace as it's already embedded in the workspace we are importing.

Reply:
That is what I understood too. Are we missing something, @sebsoto? Where is the extra API call coming from?

Reply:
I think @sebsoto is referring to the [this call in `getWorkspaceFromExistingMachineSet()`](https://github.com/openshift/windows-machine-config-operator/blob/47a9b7796075ee2f2961b4e37ad54c8f7a386bc9/test/e2e/providers/vsphere/vsphere.go#L74) despite [already getting the cluster ID here](https://github.com/openshift/windows-machine-config-operator/blob/47a9b7796075ee2f2961b4e37ad54c8f7a386bc9/test/e2e/providers/vsphere/vsphere.go#L120).

Reply:
@saifshaikh48 @sebsoto got it, I updated the PR to prevent the additional call.